### ISSUE TITLE
Add ability to remap key bindings

### ISSUE THREAD

**Issue by oreid-zd** [lusingander/stu#39]
Would be great to be able to remap key bindings via environment variables or something.

Thanks!

**Comment by lusingander** [comment]
Yes, I agree that this feature is important. I'd like to implement it, but I haven't started yet.

Related: https://github.com/lusingander/stu/issues/25

**Comment by oreid-zd** [comment]
Thanks @lusingander! I haven't written much Rust but could give it a crack if you'd accept a PR?

**Comment by lusingander** [comment]
Thanks for showing interest in this project!

I think it might help to take a look at the custom keybinding implementations from another TUI project of mine:

- https://github.com/lusingander/serie/blob/master/src/keybind.rs
- https://github.com/lusingander/serie/issues/15

Basically, I think the implementation should follow a similar approach. But there are a few things to keep in mind:

- It should be possible to assign multiple keybindings to a single action.
- It should also be possible to assign the same keybinding to multiple actions.
  - If keybindings overlap in the same view state, it’s okay as long as one is chosen internally.

If you're planning to submit a PR, it might be a good idea to split it into two steps:

- First, update the code to fetch fixed keybindings from variables.
- Then, add support for loading custom keybindings from `config.toml` or `keybindings.toml`, so users can set their own.

Feel free to reach out if you have any questions or need help along the way.

**Comment by lusingander** [comment]
https://github.com/lusingander/stu/releases/tag/v0.7.0

**Comment by oreid-zd** [comment]
Thanks legend!
